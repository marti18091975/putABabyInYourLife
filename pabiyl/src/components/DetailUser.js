import React, { useState, useEffect } from 'react';
//import { Prompt } from 'react-router-dom';
//import user from '../stores/UserStore';
import userStore from '../stores/UserStore';
import { loadDetails } from '../actions/detailAction';


import './detailUser.css';

function DetailUser(props) {
    const detailUserId = '5f4e6fc673d494545cfbadfc';
    const [detailUsers, setDetailUsers] = useState(userStore.getDetailUsers());
    const [detailElement, setDetailElement] = useState({});
    // [user,setUser] = useState({})
    const [name, setName] = useState();
    const [gender, setGender] = useState();
    const [age, setAge] = useState();
    const [city, setCity] = useState();
    const [country, setCountry] = useState();
    const [civilState, setCivilState] = useState();
    const [sons, setSons] = useState();
    const [job, setJob] = useState();
    const [presentation, setPresentation] = useState();
    const [mainImage, setMainImage] = useState();
    const [secondImage, setSecondImage] = useState();
    const [thirdImage, setThirdImage] = useState();
    // const [isFormDirty, setIsFormDirty] = useState(false);


    useEffect(() => {

        userStore.addChangeListener(onChange);
        debugger
        const detailUserId = '5f4e6fc673d494545cfbadfc';
        if (detailUsers.length === 0) {
            debugger
            console.log("something is going wrong");
            loadDetails();
        } else if (detailUserId) {
            const detailUser = userStore.getDetailUserById(detailUserId);
            if (detailUser) {
                setDetailElement(detailUser);
                setName(detailUser.name);
                setGender(detailUser.gender);
                setAge(detailUser.age);
                setCity(detailUser.city);
                setCountry(detailUser.country);
                setCivilState(detailUser.civilState);
                setSons(detailUser.sons);
                setJob(detailUser.job);
                setPresentation(detailUser.presentation);
                setMainImage(detailUser.mainimage);
                setSecondImage(detailUser.secondimage);
                setThirdImage(detailUser.thirdimage);
            }
            debugger
        }
        return () => userStore.removeChangeListener(onChange);

    }, [detailUsers.length, detailElement, detailUserId, onChange]);

    function onChange() {
        setDetailUsers(userStore.getDetailUsers());

    }

    return (
        <main>
            <h1 className="title">{name} PROFILE</h1>
            <div className="center">
                <div className="center__left">
                    <section className="left-side">
                        <div className="text">Name: {name}</div>
                        <div className="text">Gender: {gender}</div>
                        <div className="text">Age: {age}</div>
                        <div className="text">City: {city}</div>
                        <div className="text">Country: {country}</div>
                        <div className="text">Civil state: {civilState}</div>
                        <div className="text">Sons: {sons}</div>
                        <div className="text">Job: {job}</div>
                        <div className="buttons">
                            <button className="buttonE-mail"></button>
                            <button className="buttonChat"></button>
                        </div>
                    </section>
                    <section className="center-side">
                        <img className="image" src={mainImage} alt=""></img>
                    </section>
                </div>
                <section className="right-side">
                    <img src={secondImage} alt=""></img>
                    <img src={thirdImage} alt=""></img>
                </section>
            </div>
            <div className="presentation">{presentation}</div>

        </main>
    )
}
export default DetailUser;