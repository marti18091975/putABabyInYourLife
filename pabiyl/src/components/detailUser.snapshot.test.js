import renderer from 'react-test-renderer';
import React from 'react';
import DetailUser from './DetailUser';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {
    const props = {
        match: {
            params: {
                detailUserId: "5f5dcd1985b8254bf0e90f23"
            }
        }
    };
    const tree = renderer.create(
        <Router>
            <DetailUser {...props} />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});