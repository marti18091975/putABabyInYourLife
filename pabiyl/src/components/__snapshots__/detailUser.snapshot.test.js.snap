// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<main
  className="main__detail"
>
  <div
    className="main__detail--top"
  >
    <div
      className="left-side"
    >
      <section
        className="container__info"
      >
        <div
          className="text__detail name__detail"
        >
           
        </div>
        <div
          className="container__bottom--detail"
        >
          <div
            className="container"
          >
            <div
              className="container__detail"
            >
              <img
                alt="gender"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/3260/3260246.svg"
              />
              <div
                className="text__detail"
              />
            </div>
            <div
              className="container__detail"
            >
              <img
                alt="age"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/3021/3021790.svg"
              />
              <div
                className="text__detail"
              />
            </div>
          </div>
          <div
            className="container"
          >
            <div
              className="container__detail"
            >
              <img
                alt="City"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/888/888063.svg"
              />
              <div
                className="text__detail"
              />
            </div>
            <div
              className="container__detail"
            >
              <img
                alt="country"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/3455/3455212.svg"
              />
              <div
                className="text__detail"
              />
            </div>
          </div>
          <div
            className="container group-three"
          >
            <div
              className="container__detail"
            >
              <img
                alt="civil state"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/1102/1102457.svg"
              />
              <div
                className="text__detail"
              />
            </div>
            <div
              className="container__detail"
            >
              <img
                alt="sons"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/1761/1761430.svg"
              />
              <div
                className="text__detail"
              />
            </div>
            <div
              className="container__detail"
            >
              <img
                alt="job"
                className="icon__detail"
                src="https://image.flaticon.com/icons/svg/1063/1063196.svg"
              />
              <div
                className="text__detail"
              />
            </div>
          </div>
        </div>
        <a
          aria-current={null}
          className="buttons bt__inside"
          href="/emailEdit/undefined"
          onClick={[Function]}
        >
          <button
            className="buttonE-mail"
          />
        </a>
      </section>
      <section
        className="left-side__bottom"
      >
        <img
          alt=""
          className="image__secundary"
        />
        <img
          alt=""
          className="image__secundary"
        />
      </section>
    </div>
    <section
      className="right-side__detail"
    >
      <img
        alt=""
        className="image__main"
      />
      <a
        aria-current={null}
        className="buttons bt__outside"
        href="/emailEdit/undefined"
        onClick={[Function]}
      >
        <button
          className="buttonE-mail--outside"
        />
      </a>
    </section>
  </div>
  <div
    className="presentation"
  />
</main>
`;
